contracts/clearing/TradingApp.sol

```diff
@@ -92,46 +61,105 @@ contract TradingApp is IForceMoveApp {
 
 		// order
 		if (candTurnNum % 2 == 0) {
-			ITradingStructs.Order memory prevOrder = abi.decode(
+			ITradingTypes.Order memory prevOrder = abi.decode(
 				proof0.appData,
-				(ITradingStructs.Order)
+				(ITradingTypes.Order)
 			);
-			ITradingStructs.OrderResponse memory prevOrderResponse = abi.decode(
+			ITradingTypes.OrderResponse memory prevOrderResponse = abi.decode(
 				proof1.appData,
-				(ITradingStructs.OrderResponse)
+				(ITradingTypes.OrderResponse)
 			);
-			if (prevOrderResponse.responseType == ITradingStructs.OrderResponseType.ACCEPT) {
+			if (prevOrderResponse.responseType == ITradingTypes.OrderResponseType.ACCEPT) {
 				require(
 					prevOrderResponse.orderID == prevOrder.orderID,
 					'orderResponse.orderID != prevOrder.orderID, candidate is order'
 				);
 			}
 			// NOTE: used just to check the data structure validity
-			ITradingStructs.Order memory _candOrder = abi.decode(
+			ITradingTypes.Order memory _candOrder = abi.decode(
 				candidateData,
-				(ITradingStructs.Order)
+				(ITradingTypes.Order)
 			);
 			return (true, '');
 		}
 
 		// orderResponse
 		// NOTE: used just to check the data structure validity
-		ITradingStructs.OrderResponse memory _prevOrderResponse = abi.decode(
+		ITradingTypes.OrderResponse memory _prevOrderResponse = abi.decode(
 			proof0.appData,
-			(ITradingStructs.OrderResponse)
+			(ITradingTypes.OrderResponse)
 		);
 
-		ITradingStructs.Order memory order = abi.decode(proof1.appData, (ITradingStructs.Order));
-		ITradingStructs.OrderResponse memory orderResponse = abi.decode(
+		ITradingTypes.Order memory order = abi.decode(proof1.appData, (ITradingTypes.Order));
+		ITradingTypes.OrderResponse memory orderResponse = abi.decode(
 			candidateData,
-			(ITradingStructs.OrderResponse)
+			(ITradingTypes.OrderResponse)
 		);
-		if (orderResponse.responseType == ITradingStructs.OrderResponseType.ACCEPT) {
+		if (orderResponse.responseType == ITradingTypes.OrderResponseType.ACCEPT) {
 			require(
 				orderResponse.orderID == order.orderID,
 				'orderResponse.orderID != order.orderID, candidate is orderResponse'
 			);
 		}
 		return (true, '');
 	}
+
+	function verifyProofForSettlement(
+		ITradingTypes.Settlement memory settlement,
+		RecoveredVariablePart[] calldata proof
+	) internal pure {
+		if (proof.length == 1) {
+			VariablePart memory proof = proof[0].variablePart;
+			ITradingTypes.Order memory order = abi.decode(proof.appData, (ITradingTypes.Order));
+			return;
+		}
+
+		bytes32[] memory orderIDs = new bytes32[](proof.length);
+		for (uint256 i = 0; i < proof.length - 1; i++) {
+			VariablePart memory currProof = proof[i].variablePart;
+			VariablePart memory nextProof = proof[i + 1].variablePart;
+
+			require(currProof.turnNum + 1 == nextProof.turnNum, 'turns are not consecutive');
```

Comment:
We can safely assume that the order turn num should start with 2, its response 3 and so on, as all previous states are provided in proof for Settlement.